subsys/bluetooth/audio/has.c

```diff
@@ -19,13 +19,45 @@
 #define LOG_MODULE_NAME bt_has
 #include "common/log.h"
 
+/* HAS requires encryption key with at least 128 bits of entropy */
+#define HAS_ENC_KEY_SIZE_MIN 16
+
 static struct bt_has has;
 
+static int check_security(struct bt_conn *conn)
+{
+	/* The security requirements for all characteristics is LE Security Mode 1 Level 2 */
+	BT_ASSERT(bt_conn_get_security(conn) > BT_SECURITY_L1);
+
+	if (bt_conn_enc_key_size(conn) < HAS_ENC_KEY_SIZE_MIN) {
+		return BT_GATT_ERR(BT_ATT_ERR_ENCRYPTION_KEY_SIZE);
+	}
```

Comment:
Unfortunately since the security requirements for LE Audio does not match a security level, we can't just do `bt_conn_get_security` and do a check on the appropriate level. 

Due to this we'll need to implement a new function to return whether on not it is a secure connection. 
For Zephyr I don't think we need to worry about the BR/EDR. 

Not sure what do to with the `OOB method` verification.

Reply:
I guess your comment address this line:
```
	/* The security requirements for all characteristics is LE Security Mode 1 Level 2 */
	BT_ASSERT(bt_conn_get_security(conn) > BT_SECURITY_L1);
```
This assert ensures the characteristic if BT_GATT_PERM_READ_ENCRYPT and/or BT_GATT_PERM_WRITE_ENCRYPT property bit is set and the characteristic is not accessed if there's no encryption.
The LE Security Mode 1 Level 2 requirement is defined in the specification so the match is valid even if more security requirements are defined.

Regarding `bt_conn_enc_key_size()` check is used to validate the encryption key size that shall be at least 128bit as defined is specification.

Reply:
That is true, but I was referring to LE Audio basically using something like security level 2.5 (i.e. security level 2 + 128-bit key requirement and LE Secure Connections/OOB). 

Unless we can somehow move this permission check to GATT, how do you feel about introducing two new functions, 

`bt_conn_is_secure_connection(conn)` and `bt_conn_is_oob_secured(conn)` so that we can easily verify that it is either encrypted with LE Secure Connections, or by OOB? (`bt_conn_is_secure_connection` is probably poor choice, since it may give the wrong impression on people unfamiliar with LE secure connections).